app/scripts/controllers/permissions/background-api.test.js

```diff
@@ -882,26 +1023,21 @@ describe('permission background API methods', () => {
 
       getPermissionBackgroundApiMethods({
         permissionController,
-      }).removePermittedChain('foo.com', '0xdeadbeef');
+      }).removePermittedChain('foo.com', 'eip155:12345');
 
       expect(permissionController.updateCaveat).not.toHaveBeenCalled();
       expect(permissionController.revokePermission).not.toHaveBeenCalled();
     });
 
-    it('revokes the entire permission if the removed chain is the only eip:155 scope', () => {
+    // TODO: Verify this behavior (wallet vs non-wallet scopes)
+    it('revokes the entire permission if the removed chain is the only scope', () => {
```

Comment:
awkwardness here with `wallet` and `wallet:eip155` also since they're considered equal important as other chains with the way things are written now.

Reply:
This may not matter because the UI force you to explicitly disconnect if all chains are unselected